src/base/tests/test_util.py

```diff
@@ -1424,13 +1424,24 @@ def test_expand_braces_failure(self, value):
             for value in (
                 [
                     "a",
-                    "a.b" "a.b()",
+                    "a.b",
+                    "a.b()",
                     "a.b(c)",
+                    "a[b]",
+                    "context['company_id']",
                     "[('company_id', 'in', company_ids)]",
                     "[]",
                 ]
-                + ["a {} 4".format(op) for op in ["+", "-", "*", "/", "//", "%"]]
-                + ["4 {} b".format(op) for op in ["+", "-", "*", "/", "//", "%"]]
+                + ["(a {} 4)".format(op) for op in ["+", "-", "*", "/", "//", "%", "**"]]
+                + ["(4 {} b)".format(op) for op in ["+", "-", "*", "/", "//", "%", "**"]]
+                + [
+                    "((a + b) * c)",
+                    "(a + (b * c))",
+                    "((a + ((b / c) - d)) % e)",
+                    "+(a)",
+                    "-(a)",
+                    "-((a + b))",
```

Comment:
I think we should add another tests that gets the non-parenthesized versions and ensures the parenthesized version we get back matches the expected result
Example:
```
a*-b -> (a*(-b)) # and the other examples here.
```